### PR TITLE
Option to overwrite only changed files

### PR DESCRIPTION
**Pull request by wheezil**
### PR checklist

- [X] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [X] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [X] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`~~, `3.4.x`, `4.0.x`~~. Default: `master`.
- [NA] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR

Add option --minimal-overwrite which will cause output files to be written to a temp file, compared to any existing file, and if different renamed.  On build systems (such as MSBuild) that was weak dependency rules, this can help to prevent triggering of massive recompile on every build.

This resolves conflicts with changes from other authors in which new File() was replaced by Paths.get(), I kept the Paths.get() updates where applicable.

Related PRs:  #1397, #1391

### CODE REVIEW

File: modules/openapi-generator-cli/src/main/java/org/openapitools/codegen/cmd/Generate.java [hunk 1]
```diff
@@ -345,6 +350,9 @@ public void run() {
 
         if (generateAliasAsModel != null) {
             configurator.setGenerateAliasAsModel(generateAliasAsModel);
+		}
```
Comment (wing328) [inline modules/openapi-generator-cli/src/main/java/org/openapitools/codegen/cmd/Generate.java]: @wheezil please use 4-space instead of tab

**Comment by wing328** [comment]
cc @OpenAPITools/generator-core-team

**Comment by wing328** [comment]
@wheezil thanks for the contribution. Can you also add this option to the maven plugin and gradle plugin?

**Comment by wheezil** [comment]
W00T!  My second PR :-)
@wing328 is there some RTFM for these plugins?  I've never touched them.

**Comment by wing328** [comment]
It's not difficult. Just search for another option "generateAliasAsModel" in the plugins and you will know where to add the code for your new option.

**Pull request merged**